src/quicklist.c

```diff
@@ -686,19 +771,41 @@ void quicklistDelEntry(quicklistIter *iter, quicklistEntry *entry) {
 
 /* Replace quicklist entry by 'data' with length 'sz'. */
 void quicklistReplaceEntry(quicklist *quicklist, quicklistEntry *entry,
-                           void *data, int sz) {
-    /* quicklistNext() and quicklistIndex() provide an uncompressed node */
-    entry->node->zl = ziplistReplace(entry->node->zl, entry->zi, data, sz);
-    quicklistNodeUpdateSz(entry->node);
-    quicklistCompress(quicklist, entry->node);
+                           void *data, size_t sz) {
+    if (likely(!QL_NODE_IS_PLAIN(entry->node)) && likely(!isLargeElement(sz))) {
```

Comment:
```suggestion
    if (likely(!QL_NODE_IS_PLAIN(entry->node) && !isLargeElement(sz))) {
```
It seems that the result is the same for both ways of writing.
https://godbolt.org/z/zh9WdoPqK